pycbc/waveform/generator.py

```diff
@@ -276,12 +276,21 @@ def _postgenerate(self, res):
         """Applies a taper if it is in current params.
         """
         hp, hc = res
-        try:
-            hp = taper_timeseries(hp, tapermethod=self.current_params['taper'])
-            hc = taper_timeseries(hc, tapermethod=self.current_params['taper'])
-        except KeyError:
-            pass
-        return hp, hc
+
+        if self.current_params.get('taper_method') == 'constant':
+            if self.current_params['taper_window'] is None:
+                raise ValueError("If taper_method is 'constant', taper_window must be set")
+            else:
+                gate_params_hp = [(hp.start_time, self.current_params['taper_window']/2, self.current_params['taper_window'])]
```

Comment:
See previous comment as this shouldn't be so complicated here if you simplify the method arguments.